exercises/palindrome-products/test/test_palindrome_products.c

```diff
@@ -9,24 +12,45 @@ void tearDown(void)
 {
 }
 
-void check_factors(factor_t * actual, int depth, factor_t expected[])
+bool factor_t_are_equal(const factor_t * const f1, const factor_t * const f2)
```

Comment:
Can you add comments for this group of functions explaining what they do? Especially since there's 3 of them, maybe add a block comment above with the explanation, and a comment below noting the end of the helper functions.